rhel/ubi9/Dockerfile

```
@@ -0,0 +1,112 @@
+FROM registry.access.redhat.com/ubi9/ubi:9.4-1214 as jre-build
+SHELL ["/bin/bash", "-e", "-u", "-o", "pipefail", "-c"]
+
+# This Build ARG is populated by Docker
+# Ref. https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope
+ARG TARGETPLATFORM
+
+COPY adoptium-get-jdk-link.sh /usr/bin/local/adoptium-get-jdk-link.sh
+COPY adoptium-install-jdk.sh /usr/bin/local/adoptium-install-jdk.sh
+
+ARG JAVA_VERSION=17.0.12_7
+
+RUN dnf install --disableplugin=subscription-manager --setopt=install_weak_deps=0 --setopt=tsflags=nodocs --allowerasing -y \
+    ca-certificates \
+    curl \
+    jq \
+    && dnf clean --disableplugin=subscription-manager all \
+    && /usr/bin/local/adoptium-install-jdk.sh
+
+ENV PATH="/opt/jdk-${JAVA_VERSION}/bin:${PATH}"
+
+# Generate smaller java runtime without unneeded files
+# for now we include the full module path to maintain compatibility
+# while still saving space (approx 200mb from the full distribution)
+RUN case "$(jlink --version 2>&1)" in \
+      # jlink version 11 has less features than JDK17+
+      "11."*) set -- "--strip-debug" "--compress=2" ;; \
+      "17."*) set -- "--strip-java-debug-attributes" "--compress=2" ;; \
+      # the compression argument is different for JDK21
+      "21."*) set -- "--strip-java-debug-attributes" "--compress=zip-6" ;; \
+      *) echo "ERROR: unmanaged jlink version pattern" && exit 1 ;; \
+    esac; \
+    jlink \
+      "$1" \
+      "$2" \
+      --add-modules ALL-MODULE-PATH \
+      --no-man-pages \
+      --no-header-files \
+      --output /javaruntime
+
+FROM registry.access.redhat.com/ubi9/ubi:9.4-1214 AS agent
```

Similar version update as earlier.
```suggestion
FROM registry.access.redhat.com/ubi9/ubi:9.4-1214.1726694543 AS agent
```